### PR TITLE
Piwigo

### PR DESCRIPTION
**Pull request by Narcolapser**
**What this PR does / why we need it**:

Add's Piwigo Photo Gallery.

### CODE REVIEW

File: roles/piwigo/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+piwigo_enabled: false
+piwigo_available_externally: "false"
+
+# directories
+piwigo_config_directory: "{{ docker_home }}/piwigo/config"
+piwigo_data_directory: "{{ docker_home }}/piwigo/data"
+piwigo_photos: "{{ photos_root }}/piwigo"
+
+# uid / gid
+piwigo_user_id: "1000"
```
[the remark this answers is not in the document]
Comment (Narcolapser) [reply roles/piwigo/defaults/main.yml:11]: Apparently not, quoth https://hub.docker.com/r/linuxserver/piwigo

```
docker run -d \
  --name=piwigo \
  -e PUID=1000 \
  -e PGID=1000 \
  -e TZ=Europe/London \
  -p 80:80 \
  -v </path/to/appdata/config>:/config \
  -v </path/to/appdata/gallery>:/gallery \
  --restart unless-stopped \
  ghcr.io/linuxserver/piwigo
```
It does seem strange though. I just logged into the container and it is running as root, but the nginx and php processes are running as user "abc" which is UID of 1000. Can never have to much security I guess?

**Comment by Narcolapser** [comment]
Just reminding you this is still dandling. The uid/gid thing appears to be how the container maintainers intended it. So I stuck with their guidance. It wouldn't be the first container to do such, currently the grafana container uses the "grafana" user which is 472 in my container. In this case it appears to be running the service as the user "abc".

**Comment by Narcolapser** [comment]
I ran across something else a while back that explained gid and uid, I agree, those should have been 0. I have updated them so this should be ready to merge now.

**Comment by seeReadCode** [comment]
My experience: using `0` for the PUID and PGID generates a 502 and this error: `please specify user and group other than root`. Using `1000` allows piwigo to actually load.

**Pull request merged**